### DISCUSSION TITLE
Hosted app isn't authenticating to github.com correctly

### DISCUSSION THREAD

**Discussion by mikebryant** [renovatebot/renovate#10334] [Request Help]
I'm using the hosted app, this particular run is https://app.renovatebot.com/dashboard#github/BulbEnergy/argocd-root/371774361

The private repository has ArgoCD dependencies on other private repositories within the same organisation.

Error is:
```
DEBUG: Found argocd package files
DEBUG: Found 42 package file(s)
INFO: Dependency extraction complete
{
  "baseBranch": "master",
  "stats": {
    "managers": {
      "argocd": {
        "fileCount": 42,
        "depCount": 44
      }
    },
    "total": {
      "fileCount": 42,
      "depCount": 44
    }
  }
}
DEBUG: Datasource unknown error
{
  "datasource": "git-tags",
  "lookupName": "https://github.com/BulbEnergy/cluster-base.git",
  "err": {
    "task": {
      "commands": [
        "ls-remote",
        "https://github.com/BulbEnergy/cluster-base.git"
      ],
      "format": "utf-8"
    },
    "message": "fatal: could not read Username for 'https://github.com': No such device or address\n",
    "stack": "Error: fatal: could not read Username for 'https://github.com': No such device or address\n\n    at Object.action (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/plugins/error-detection.plugin.js:30:33)\n    at PluginStore.exec (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/plugins/plugin-store.js:21:33)\n    at /home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:95:45\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:93:16)\n    at GitExecutorChain.<anonymous> (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:77:46)\n    at Generator.next (<anonymous>)\n    at fulfilled (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
  }
}
DEBUG: Failed to look up dependency https://github.com/BulbEnergy/cluster-base.git (https://github.com/BulbEnergy/cluster-base.git)(packageFile="objects/bulb-ci-prod/europe-west2/ci-primary/apps/bulbenrg-cluster-base.yaml", dependency="https://github.com/BulbEnergy/cluster-base.git")
```

I was expecting Renovate to have access to this repository - indeed, hosted Renovate is currently running correctly against https://github.com/BulbEnergy/cluster-base.git

But when it tries to do the listing of references, it doesn't appear to be populating the token

The file it's failing on (eliding other config), is:
```
spec:
  project: default
  source:
    repoURL: https://github.com/BulbEnergy/cluster-base.git
    targetRevision: 1.0.1
````

**Comment by mikebryant** [comment]
Going by the lack of this log line: https://github.com/renovatebot/renovate/blob/main/lib/util/git/url.ts#L16
Looks like automated hostrules for github.com aren't being set?

**Comment by rarkins** [comment]
I don't think that the `git-tags` datasource supports authentication, while `github-tags` does. Possible solutions:
- Use some type of git `insteadOf` command
- Modify the https URL to embed token into it
- Detect known github hosts and redirect the query to `github-tags` datasource

**Reply by mikebryant** [reply]
It looks like it's supposed to support authentication?
https://github.com/renovatebot/renovate/blob/main/lib/datasource/git-tags/index.ts#L11 delegates to git-refs
https://github.com/renovatebot/renovate/blob/main/lib/datasource/git-refs/index.ts#L32 calls `getRemoteUrlWithToken(lookupName)`
Then that looks at host rules: https://github.com/renovatebot/renovate/blob/main/lib/util/git/url.ts#L14

As I understand it the platform code is supposed to fill in host rules here: https://github.com/renovatebot/renovate/blob/main/lib/platform/index.ts#L125

Does the hosted whitesource version do something differently perhaps?

> Use some type of git insteadOf command

I don't see any references to that in https://docs.renovatebot.com/configuration-options/? Not sure how to configure that on hosted Renovate

> Modify the https URL to embed token into it

I think I have to rule that one out on security policy grounds, not supposed to embed keys into our source repositories

> Detect known github hosts and redirect the query to github-tags datasource

I could have a go at a PR for that if you want - though since the authentication in https://github.com/renovatebot/renovate/blob/main/lib/util/git/url.ts#L14 appears to be broken, it's not clear to me why github-tags would work any better?

**Reply by rarkins** [reply]
Sorry, the options I mentioned were all in the context of things we could add to the source code of Renovate, not things you could already configure.

I checked the app logic and you're right - it was adding `api.github.com` but I've changed that to `github.com` now and it will be live in a few minutes.

**Reply by mikebryant** [reply]
Oh, right. Sorry, misunderstood

Ahh. That makes sense. Awesome, I'll keep checking back - looks like it's still not working in the latest run on my repo ~35 minutes ago

**Reply by mikebryant** [reply]
I'm afraid it looks like it's still breaking in the same way - doesn't seem to be seeing the host rule
